### ISSUE TITLE
struct pointers should be nil if no field ist set

### ISSUE THREAD

**Issue by Bios-Marcel** [Bios-Marcel/yagcl#1]
Test:

```go
func Test_Parse_SinglePointerToStruct_NoFieldsSet(t *testing.T) {
	type substruct struct {
		FieldC int `key:"field_c"`
	}
	type configuration struct {
		FieldA string     `key:"field_a"`
		FieldB *substruct `key:"field_b"`
	}

	defer setEnvTemporarily("FIELD_A", "content a")()
	var c configuration
	err := yagcl.New[configuration]().AddSource(Source()).Parse(&c)
	if assert.NoError(t, err) {
		assert.Equal(t, "content a", c.FieldA)
		assert.Equal(t, nil, c.FieldB)
	}
}
```

**Comment by Bios-Marcel** [comment]
Implement for the JSON parser with https://github.com/Bios-Marcel/yagcl-json/commit/c0a33c97ab91fe76e93a8002e814dc48ce96b124

**Comment by Bios-Marcel** [comment]
Already implemented for ENV parser, as confirmed with 07573ae56b8b23642de6b8aa00706d69224a255e
